### PR TITLE
adapt to upstream handling

### PR DESCRIPTION
**Pull request by mlilien**
- cicd: add regression test
- **breaking change**:  rm `GitVersion_BranchVersion` and `GitVersion_BranchVersionDockerLabel`
- update gitversion config
    - rename master -> main for main branch
    - reconfigure mode of feature branch to ContinuousDeployment
- entrypoint.sh: add user only if not already existing

### CODE REVIEW

File: ci/pipeline.yaml [hunk 1]
```diff
@@ -105,4 +105,63 @@ jobs:
...
+          tag_file: gitversion/plain/FullSemVerDockerLabel
+
+      - task: regression-test
+        image: image-((bb_name))
+        config:
+          platform: linux
+          inputs:
+            - name: source
+          outputs:
+            - name: gitversion
```
Comment (JoergZeidler) [inline ci/pipeline.yaml]: are input and output folder necessary for the regression-test?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,15 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.9.0] - 2021.Q3
+
+- cicd: add regression test
+- **breaking change**:  rm `GitVersion_BranchVersion` and `GitVersion_BranchVersionDockerLabel`
```
Comment (JoergZeidler) [inline CHANGELOG.md:11]: why it is removed this feature?

Comment (mlilien) [reply CHANGELOG.md:11]: as you can see in the regression test, it's obsolete. due to upstream changes the branch name of a feature branch is part of FullSemVer anyway. `BranchVersion*` would include the branch name twice. This is a potential problem for docker tags, which are limited to 128 characters.

File: ci/pipeline.yaml (hunk 1 above)
Comment (mlilien) [reply ci/pipeline.yaml]: removed

**Review by JoergZeidler — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (siredmar) [reply CHANGELOG.md:11]: Iirc BranchVersion is a functionality that determines by itself the version is generated from the default branch or any other. If generated from the default branch it shall take gitversion FullSemVer and on any other branch it shall give InformationalVersion. 
You mentioned upstream changes and that FullSemVer now includes the branch. However, there is still the '+' character that cannot be used for Docker tags. Is this covered with those midificatinons?

Comment (mlilien) [reply CHANGELOG.md:11]: You can use `FullSemVerDockerLabel` for that.

**Pull request merged**